### ISSUE TITLE
Non utf-8 characters in DESCRIPTION file

### ISSUE THREAD

**Issue by kellijohnson-NOAA** [ropensci-review-tools/pkgcheck#272]
https://github.com/ropensci-review-tools/pkgcheck/blob/c0beabcdf4c23efaccd3fa3d674164fc06c297b8/DESCRIPTION#L7
leads to warning messages when running the tests, e.g.,
```
    checks <- make_check_data ()
    ci_out <- output_pkgchk_branch_is_master (checks)
    expect_equal (
        ci_out$summary,
        "Default GitHub branch of 'master' is not acceptable."
    )
Warning messages:
1: In grep("^[^\\s]+:$", lines, perl = TRUE, value = TRUE) :
  input string 7 is invalid UTF-8
Calls: expect_equal ... idesc_create_text -> read_dcf -> sub -> is.factor -> grep
2: In grep("^[^\\s]+:$", lines, perl = TRUE, value = TRUE) :
  input string 7 is invalid UTF-8
Calls: expect_equal ... idesc_create_text -> read_dcf -> sub -> is.factor -> grep
```

This is because there are non utf-8 characters in Maelle's name. The warnings are coming from `desc::read_dcf()` because of a call to `grep()`, i.e., `grep("^[^\\s]+:$", lines, perl = TRUE, value = TRUE)`. The warning message is given twice because read_dcf is called twice while evaluating the expectation. This warning message is not a deal breaker but it is alarming when running the tests locally, especially for a new contributor. I am not sure if we can change the name in the DESCRIPTION file to use utf-8, list the encoding for the DESCRIPTION file if not utf-8, or I need to move this issue with a reproducible example to {desc}? Thanks in advance for the guidance here.

**Comment by maelle** [comment]
👋 @kellijohnson-NOAA @mpadge! Interesting 😅 I wouldn't change my name. 

I wonder whether DESCRIPTION files ever use another Encoding (I always use UTF-8, despite listing my name). I think preparing a reprex for desc makes sense.

It reminds me of https://github.com/r-lib/desc/issues/80

**Comment by maelle** [comment]
This works on my machine (where usethis picks up my name)

```r
dir <- withr::local_tempdir()
usethis::create_package(dir, open = FALSE)
desc:::read_dcf(file.path(dir, "DESCRIPTION"))
```

where exactly is `desc:::read_dcf()` called here?

**Comment by kellijohnson-NOAA** [comment]
@maelle sorry I didn't mean to suggest that you should change your name. When the file is initially read in it looks fine, i.e.,
```
person(\"Maëlle\", \"Salmon\", role = \"aut\")
```
but after `Encoding(res) <- encoding` it looks like
```
person(\"Ma\xeblle\", \"Salmon\", role = \"aut\")
```
But, if you remove the encoding specification from the DESCRIPTION file then  you get other warnings/messages when you load the package.

Here is the call trace from `testthat::expect_equal()`
```
where 1: read_dcf(con)
where 2: idesc_create_text(self, private, lines)
where 3: idesc_create_file(self, private, file)
where 4: idesc_create(self, private, cmd, file, text, package)
where 5: initialize(...)
where 6: description$new(cmd, file, text, package)
where 7: desc_desc(pkg_path(path))
where 8: pkgload::pkg_desc(path)
where 9: doTryCatch(return(expr), name, parentenv, handler)
where 10: tryCatchOne(expr, names, parentenv, handlers[[1L]])
where 11: tryCatchList(expr, classes, parentenv, handlers)
where 12: tryCatch(pkgload::pkg_desc(path), error = function(e) NULL)
where 13: find_description(path, package)
where 14: find_edition(".")
where 15: the$edition %||% find_edition(".")
where 16: edition_get()
where 17: expect_equal(ci_out$summary, "Default GitHub branch of 'master' is not acceptable.")
```

I am happy to work out a reprex for {desc} because this issue is not actually a problem of {pkgcheck}.

**Comment by mpadge** [comment]
This all seems fine now, so I'm going to close.
